From be09d5480d54da6a9cb7428886c0c700b8a84e0b Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Thu, 2 Mar 2017 15:03:31 +0800
Subject: [PATCH 1740/5242] MLK-14314-2 dma: pxp: refine is_yuv().

commit  0434c64e16f1bf44a982bc2e2d71fcb4d2cf2b3c from
https://source.codeaurora.org/external/imx/linux-imx.git

Use 'switch' statement to replace 'if' statement to
make logic more clear and easier to maintain.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit ca1b4393b86054a81bb40ad856af9c4f166841ea)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/pxp/pxp_dma_v2.c |   45 +++++++++++++++++++++--------------------
 drivers/dma/pxp/pxp_dma_v3.c |   46 +++++++++++++++++++++---------------------
 2 files changed, 46 insertions(+), 45 deletions(-)

diff --git a/drivers/dma/pxp/pxp_dma_v2.c b/drivers/dma/pxp/pxp_dma_v2.c
index 0446bc2..710729c 100644
--- a/drivers/dma/pxp/pxp_dma_v2.c
+++ b/drivers/dma/pxp/pxp_dma_v2.c
@@ -215,28 +215,29 @@ static void dump_pxp_reg(struct pxps *pxp)
 
 static bool is_yuv(u32 pix_fmt)
 {
-	if ((pix_fmt == PXP_PIX_FMT_YUYV) |
-	    (pix_fmt == PXP_PIX_FMT_UYVY) |
-	    (pix_fmt == PXP_PIX_FMT_YVYU) |
-	    (pix_fmt == PXP_PIX_FMT_VYUY) |
-	    (pix_fmt == PXP_PIX_FMT_Y41P) |
-	    (pix_fmt == PXP_PIX_FMT_VUY444) |
-	    (pix_fmt == PXP_PIX_FMT_NV12) |
-	    (pix_fmt == PXP_PIX_FMT_NV21) |
-	    (pix_fmt == PXP_PIX_FMT_NV16) |
-	    (pix_fmt == PXP_PIX_FMT_NV61) |
-	    (pix_fmt == PXP_PIX_FMT_GREY) |
-	    (pix_fmt == PXP_PIX_FMT_GY04) |
-	    (pix_fmt == PXP_PIX_FMT_YVU410P) |
-	    (pix_fmt == PXP_PIX_FMT_YUV410P) |
-	    (pix_fmt == PXP_PIX_FMT_YVU420P) |
-	    (pix_fmt == PXP_PIX_FMT_YUV420P) |
-	    (pix_fmt == PXP_PIX_FMT_YUV420P2) |
-	    (pix_fmt == PXP_PIX_FMT_YVU422P) |
-	    (pix_fmt == PXP_PIX_FMT_YUV422P)) {
-		return true;
-	} else {
-		return false;
+	switch (pix_fmt) {
+		case PXP_PIX_FMT_YUYV:
+		case PXP_PIX_FMT_UYVY:
+		case PXP_PIX_FMT_YVYU:
+		case PXP_PIX_FMT_VYUY:
+		case PXP_PIX_FMT_Y41P:
+		case PXP_PIX_FMT_VUY444:
+		case PXP_PIX_FMT_NV12:
+		case PXP_PIX_FMT_NV21:
+		case PXP_PIX_FMT_NV16:
+		case PXP_PIX_FMT_NV61:
+		case PXP_PIX_FMT_GREY:
+		case PXP_PIX_FMT_GY04:
+		case PXP_PIX_FMT_YVU410P:
+		case PXP_PIX_FMT_YUV410P:
+		case PXP_PIX_FMT_YVU420P:
+		case PXP_PIX_FMT_YUV420P:
+		case PXP_PIX_FMT_YUV420P2:
+		case PXP_PIX_FMT_YVU422P:
+		case PXP_PIX_FMT_YUV422P:
+			return true;
+		default:
+			return false;
 	}
 }
 
diff --git a/drivers/dma/pxp/pxp_dma_v3.c b/drivers/dma/pxp/pxp_dma_v3.c
index c03d9ec..32ca050 100644
--- a/drivers/dma/pxp/pxp_dma_v3.c
+++ b/drivers/dma/pxp/pxp_dma_v3.c
@@ -393,32 +393,32 @@ static void print_param(struct pxp_layer_param *p, char *s)
 
 static bool is_yuv(u32 pix_fmt)
 {
-	if ((pix_fmt == PXP_PIX_FMT_YUYV) |
-	    (pix_fmt == PXP_PIX_FMT_UYVY) |
-	    (pix_fmt == PXP_PIX_FMT_YVYU) |
-	    (pix_fmt == PXP_PIX_FMT_VYUY) |
-	    (pix_fmt == PXP_PIX_FMT_Y41P) |
-	    (pix_fmt == PXP_PIX_FMT_VUY444) |
-	    (pix_fmt == PXP_PIX_FMT_NV12) |
-	    (pix_fmt == PXP_PIX_FMT_NV21) |
-	    (pix_fmt == PXP_PIX_FMT_NV16) |
-	    (pix_fmt == PXP_PIX_FMT_NV61) |
-	    (pix_fmt == PXP_PIX_FMT_GREY) |
-	    (pix_fmt == PXP_PIX_FMT_GY04) |
-	    (pix_fmt == PXP_PIX_FMT_YVU410P) |
-	    (pix_fmt == PXP_PIX_FMT_YUV410P) |
-	    (pix_fmt == PXP_PIX_FMT_YVU420P) |
-	    (pix_fmt == PXP_PIX_FMT_YUV420P) |
-	    (pix_fmt == PXP_PIX_FMT_YUV420P2) |
-	    (pix_fmt == PXP_PIX_FMT_YVU422P) |
-	    (pix_fmt == PXP_PIX_FMT_YUV422P)) {
-		return true;
-	} else {
-		return false;
+	switch (pix_fmt) {
+		case PXP_PIX_FMT_YUYV:
+		case PXP_PIX_FMT_UYVY:
+		case PXP_PIX_FMT_YVYU:
+		case PXP_PIX_FMT_VYUY:
+		case PXP_PIX_FMT_Y41P:
+		case PXP_PIX_FMT_VUY444:
+		case PXP_PIX_FMT_NV12:
+		case PXP_PIX_FMT_NV21:
+		case PXP_PIX_FMT_NV16:
+		case PXP_PIX_FMT_NV61:
+		case PXP_PIX_FMT_GREY:
+		case PXP_PIX_FMT_GY04:
+		case PXP_PIX_FMT_YVU410P:
+		case PXP_PIX_FMT_YUV410P:
+		case PXP_PIX_FMT_YVU420P:
+		case PXP_PIX_FMT_YUV420P:
+		case PXP_PIX_FMT_YUV420P2:
+		case PXP_PIX_FMT_YVU422P:
+		case PXP_PIX_FMT_YUV422P:
+			return true;
+		default:
+			return false;
 	}
 }
 
-
 static void pxp_set_ctrl(struct pxps *pxp)
 {
 	struct pxp_config_data *pxp_conf = &pxp->pxp_conf_state;
-- 
1.7.9.5

